model/host/host.go

```diff
@@ -380,6 +380,14 @@ func (h *Host) MarkAsProvisioned() error {
 			},
 		},
 	)
+	if err == nil {
+		h.Status = evergreen.HostRunning
+		h.Provisioned = true
+		h.ProvisionTime = time.Now()
+		return nil
+	}
+
+	return err
```

Comment:
In general it's a bit more idiomatic to return the error state first/early, and then do the "non-error" logic afterwards. Makes refactoring later easier.

Reply:
Fixed!